src/Plugin/search_api_autocomplete/suggester/BackendTrait.php

```diff
@@ -8,6 +8,10 @@
 
 /**
  * Provides a helper method for loading the search backend.
+ * @deprecated in search_api_solr:4.3.0 and is removed from search_api_solr:5.0.0. Use the
+ *   \Drupal\search_api_solr_autocomplete\Plugin\search_api_autocomplete\suggester\BackendTrait instead
+ *
+ * @see https://www.drupal.org/node/3254186
  */
 trait BackendTrait {
```

Comment:
It's not enough to add the `@deprecated` tag in order to deprecate. Please read again and carefully https://www.drupal.org/about/core/policies/core-change-policies/drupal-deprecation-policy#how-class.

Also, a deprecated class should only point to the new code and remove their code:

```php

use Drupal\search_api_solr_autocomplete\Plugin\search_api_autocomplete\suggester\BackendTrait as BackendTraitOriginal

trait BackendTrait {
  use BackendTraitOriginal;
}
```